src/shared_modules/router/tests/unit/publisher_test.cpp

```diff
@@ -108,3 +109,110 @@ TEST_F(PublisherTest, TestPublishEmptyData)
     // Check that the Publisher class can publish empty data
     EXPECT_NO_THROW(publisher->push(emptyData));
 }
+
+/*
+ * @brief Tests send data to socket without header P
+ */
+TEST_F(PublisherTest, TestPublishSocketWithoutP)
```

Comment:
This test has to much sleep time.
You could add some wait and notify logic, here is an example

<details><summary>Details</summary>
<p>


```c++
/*
 * @brief Tests send data to socket without header P
 */
TEST_F(PublisherTest, TestPublishSocketWithoutP)
{
    const std::string ENDPOINT_NAME = "test";
    const std::string SOCKET_PATH = "test/";
    std::condition_variable cv;
    std::mutex cvMutex;

    const auto publisher = std::make_shared<Publisher>(ENDPOINT_NAME, SOCKET_PATH);
    auto socketClient = std::make_unique<SocketClient<Socket<OSPrimitives>, EpollWrapper>>(SOCKET_PATH + ENDPOINT_NAME);

    nlohmann::json jsonMessage;
    jsonMessage["type"] = "subscribe";
    jsonMessage["subscriberId"] = "ID_0";
    auto jsonMessageString = jsonMessage.dump();

    int32_t onReadCallCount = 0;

    socketClient->connect(
        [&](const char* body, uint32_t bodySize, const char*, uint32_t)
        {
            std::unique_lock<std::mutex> lk(cvMutex);
            nlohmann::json result;
            EXPECT_NO_THROW(result = nlohmann::json::parse(body, body + bodySize));
            if (onReadCallCount == 0)
            {
                EXPECT_EQ(result.dump(), R"({"Result":"OK"})");
            }
            else
            {
                EXPECT_EQ(
                    result.dump(),
                    R"({"offset":57000,"paths":["GracefulShutdown.json"],"stageStatus":[{"stage":"download","status":"ok"}],"type":"offsets"})");
            }
            onReadCallCount++;
            cv.notify_all();
        },
        [&]()
        {
            EXPECT_NO_THROW(socketClient->send(jsonMessageString.data(), jsonMessageString.size()));
        });
    {
        std::unique_lock<std::mutex> lk(cvMutex);
        auto result = cv.wait_for(lk, std::chrono::seconds(5), [&]() { return onReadCallCount > 0; });

        if (!result)
        {
            FAIL() << "Timeout";
        }
    }
    auto routerMessageJson = R"(
    {
        "type": "offsets",
        "offset": 57000,
        "paths":
        [
            "GracefulShutdown.json"
        ],
        "stageStatus":
        [
            {
                "stage": "download",
                "status": "ok"
            }
        ]
    }
    )"_json;

    const auto routerMessagePayload = routerMessageJson.dump();
    const auto routerMessage = std::vector<char>(routerMessagePayload.begin(), routerMessagePayload.end());
    publisher->call(routerMessage);

    {
        std::unique_lock<std::mutex> lk(cvMutex);
        auto result = cv.wait_for(lk, std::chrono::seconds(5), [&]() { return onReadCallCount > 1; });

        if (!result)
        {
            FAIL() << "Timeout";
        }
    }
    EXPECT_EQ(onReadCallCount, 2);
}
```

</p>
</details> 

There is also an expect for `onReadCallCount` in case no messages are received.

Reply:
Done